platforms/allwinner-d1/boards/lichee-rv/src/main.rs

```diff
@@ -196,58 +199,41 @@ fn main() -> ! {
     })
     .unwrap();
 
-    // Initialize SerialMux
-    let mux_done = k
-        .initialize(async move {
-            loop {
-                // Now, right now this is a little awkward, but what I'm doing here is spawning
-                // a new virtual mux, and configuring it with:
-                // * Up to 16 virtual ports max
-                // * Framed messages up to 512 bytes max each
-                match SerialMuxServer::register(k, 16, 512).await {
-                    Ok(_) => break,
-                    Err(RegistrationError::SerialPortNotFound) => {
-                        // Uart probably isn't registered yet. Try again in a bit
-                        k.sleep(Duration::from_millis(10)).await;
-                    }
-                    Err(e) => {
-                        panic!("uhhhh {e:?}");
-                    }
-                }
-            }
-        })
-        .unwrap();
+    // Initialize the SerialMuxServer
+    k.initialize({
+        const PORTS: usize = 16;
+        const FRAME_SIZE: usize = 512;
+        async {
+            // * Up to 16 virtual ports max
+            // * Framed messages up to 512 bytes max each
+            tracing::debug!("initializing SerialMuxServer...");
+            SerialMuxServer::register(k, PORTS, FRAME_SIZE)
+                .await
+                .unwrap();
+            tracing::info!("SerialMuxServer initialized!");
+        }
+        .instrument(tracing::info_span!(
+            "SerialMuxServer",
+            ports = PORTS,
+            frame_size = FRAME_SIZE
+        ))
+    })
+    .unwrap();
 
     // initialize tracing
     k.initialize(async move {
-        mux_done.await.unwrap();
         COLLECTOR.start(k).await;
         trace::info!("started tracing");
     })
     .unwrap();
 
-    // Loopback on virtual port zero
-    k.initialize(async move {
-        let hdl = PortHandle::open(k, 0, 1024).await.unwrap();
-
-        loop {
-            let rx = hdl.consumer().read_grant().await;
-            let all_len = rx.len();
-            hdl.send(&rx).await;
-            rx.release(all_len);
-        }
-    })
-    .unwrap();
-
-    k.initialize(async move {
-        let hdl = PortHandle::open(k, 1, 1024).await.unwrap();
+    // Spawn a loopback port
+    let loopback_settings = SermuxLoopbackSettings::default();
+    k.initialize(sermux_loopback(k, loopback_settings)).unwrap();
 
-        loop {
-            hdl.send(b"Hello, world!\r\n").await;
-            k.sleep(Duration::from_secs(1)).await;
-        }
-    })
-    .unwrap();
+    // Spawn a hello port
+    let hello_settings = SermuxHelloSettings::default();
+    k.initialize(sermux_hello(k, hello_settings)).unwrap();
```

Comment:
nit, not a big deal: maybe these should go in a `services::sermux` module, so we can reference them as `sermux::loopback` and sermux::hello`?